### PR TITLE
[test] Enable HTTP server on Android

### PR DESCRIPTION
**Pull request by jmalanen**
Fixes: mapbox/mapbox-gl-native-team/issues/167

### CODE REVIEW

File: platform/android/src/run_loop.cpp [hunk 1]
```diff
@@ -243,12 +252,28 @@ void RunLoop::stop() {
     });
 }
 
-void RunLoop::addWatch(int, Event, std::function<void(int, Event)>&&) {
-    throw std::runtime_error("Not implemented.");
+void RunLoop::addWatch(int fd, Event event, std::function<void(int, Event)>&& cb) {
```
Comment (alexshalamov) [inline platform/android/src/run_loop.cpp]: Where new addWatch and removeWatch is used?

Comment (jmalanen) [reply platform/android/src/run_loop.cpp]: In platform/default HTTPFileSource::Impl::handleSocket().

File: platform/android/src/test/test_runner.cpp [hunk 2]
```diff
@@ -23,6 +25,7 @@ void runner() {
     int status = mbgl::runTests(argv.size(), argv.data());
     mbgl::Log::Info(mbgl::Event::General, "TestRunner finished with status: '%d'", status);
     running = false;
+    success = status ? false : true;
```
Comment (pozdnyakov) [inline platform/android/src/test/test_runner.cpp]: nit: success = (status == 0);

File: platform/android/src/run_loop.cpp [hunk 3]
```diff
@@ -55,6 +55,15 @@ int looperCallbackDefault(int fd, int, void* data) {
     return 1;
 }
 
+int looperCallbackReadEvent(int fd, int, void* data) {
+    auto runLoopImpl = reinterpret_cast<RunLoop::Impl*>(data);
```
Comment (pozdnyakov) [inline platform/android/src/run_loop.cpp]: is it possible that `looperCallbackReadEvent()` is invoked after `RunLoop::Impl` got deleted?

File: platform/android/src/run_loop.cpp [hunk 4]
```diff
@@ -55,6 +55,15 @@ int looperCallbackDefault(int fd, int, void* data) {
...
+
+    if (runLoopImpl->readPoll.count(fd)) {
```
Comment (pozdnyakov) [inline platform/android/src/run_loop.cpp]: nit: 
```
 auto it = impl->readPoll.find(fd);
    if (it != impl->readPoll.end()) {
```

File: platform/android/src/run_loop.cpp [hunk 5]
```diff
@@ -55,6 +55,15 @@ int looperCallbackDefault(int fd, int, void* data) {
...
+        runLoopImpl->readPoll[fd](fd, RunLoop::Event::Read);
```
Comment (pozdnyakov) [inline platform/android/src/run_loop.cpp]: assertion that the contained std::function contains valid target

File: platform/android/src/run_loop.cpp (hunk 4 above)
Comment (jmalanen) [reply platform/android/src/run_loop.cpp]: Done

File: platform/android/src/run_loop.cpp (hunk 5 above)
Comment (jmalanen) [reply platform/android/src/run_loop.cpp]: Done

File: platform/android/src/test/test_runner.cpp (hunk 2 above)
Comment (jmalanen) [reply platform/android/src/test/test_runner.cpp]: Done

File: platform/android/src/run_loop.cpp [hunk 6]
```diff
@@ -55,6 +55,17 @@ int looperCallbackDefault(int fd, int, void* data) {
     return 1;
 }
 
+int looperCallbackReadEvent(int fd, int, void* data) {
+    auto runLoopImpl = reinterpret_cast<RunLoop::Impl*>(data);
+
+    auto it = runLoopImpl->readPoll.find(fd);
+    if (it != runLoopImpl->readPoll.end()) {
+        assert(it->second != nullptr);
```
Comment (pozdnyakov) [inline platform/android/src/run_loop.cpp]: nit: `assert(it->second);`

Comment (jmalanen) [reply platform/android/src/run_loop.cpp]: Done

File: platform/android/src/run_loop.cpp (hunk 3 above)
Comment (jmalanen) [reply platform/android/src/run_loop.cpp]: Should not be possible since the HTTPFileSource and curl is cleaned up and deleted before RunLoop deletion.

**Pull request merged**